tests/mocks/reentrancy_mock.cairo

```diff
@@ -0,0 +1,132 @@
+# SPDX-License-Identifier: MIT
+
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from starkware.cairo.common.math_cmp import is_le
+from openzeppelin.utils.constants import TRUE
+from starkware.cairo.common.uint256 import (
+    Uint256,
+    uint256_signed_nn_le,
+    uint256_sub
+)
+from starkware.starknet.common.syscalls import (
+    get_contract_address
+)
+from openzeppelin.security.reentrancy_guard import (  
+    ReentrancyGuard_start,
+    ReentrancyGuard_end
+)
+
+@contract_interface
+namespace IReentrancyGuardAttacker:
+    func call_sender():
+    end
+end
+
+@contract_interface
+namespace IReentrancyGuard:
+    func count_this_recursive(n : Uint256, recursive_jump: Uint256):
+    end
+end
+
+@storage_var
+func counter() -> (res : felt):  
+end
+
+@constructor
+func constructor{
+        syscall_ptr : felt*, 
+        pedersen_ptr : HashBuiltin*,
+        range_check_ptr
+    }(initial_number: felt):
+    counter.write(initial_number)
+    return ()
+end
+
+@view
+func current_count{
+        syscall_ptr : felt*, 
+        pedersen_ptr : HashBuiltin*,
+        range_check_ptr
+    }() -> (res: felt):
+    let (res) = counter.read()
+    return (res)
+end
+
+@external
+func callback{
+        syscall_ptr : felt*, 
+        pedersen_ptr : HashBuiltin*,
+        range_check_ptr
+    }():
+   ReentrancyGuard_start()
+   _count()
+   ReentrancyGuard_end()
+   return ()
+end
+
+@external
+func count_local_recursive {
+        syscall_ptr : felt*, 
+        pedersen_ptr : HashBuiltin*,
+        range_check_ptr
+    } (n : felt):
+    alloc_locals
+    ReentrancyGuard_start()
+    let (greater_zero) = is_le(1, n)
+    if greater_zero == TRUE:           
+        _count()
+        count_local_recursive(n - 1)
+        tempvar syscall_ptr=syscall_ptr
+        tempvar pedersen_ptr=pedersen_ptr
+        tempvar range_check_ptr=range_check_ptr
+    else:
+        tempvar syscall_ptr=syscall_ptr
+        tempvar pedersen_ptr=pedersen_ptr
+        tempvar range_check_ptr=range_check_ptr
+    end
+    ReentrancyGuard_end()
+   return ()
+end
+
+@external
+func count_this_recursive {
+        syscall_ptr : felt*, 
+        pedersen_ptr : HashBuiltin*,
+        range_check_ptr
+   } (n : Uint256, recursive_jump: Uint256):
```

Comment:
echoing Andrew's question, I don't think there's a benefit in using uint256 if the recursion stays the same. 
Similarly, I'm not convinced that the recursive jump helps either.